hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/TestDataTieringManager.java

```diff
@@ -0,0 +1,378 @@
+    // Test with another valid path
+    hFilePath = hStoreFiles.get(3).getPath();
+    testDataTieringMethodWithPathNoException(methodCallerWithPath, hFilePath, false);
+
+    // Test with an incorrect filename
+    hFilePath = new Path(hStoreFiles.get(0).getPath().getParent(), "incorrectFileName");
+    testDataTieringMethodWithPathExpectingException(methodCallerWithPath, hFilePath,
+      new DataTieringException("HStoreFile corresponding to " + hFilePath + " doesn't exist"));
+  }
+
+  private void testDataTieringMethodWithPath(DataTieringMethodCallerWithPath caller, Path path,
+    boolean expectedResult, DataTieringException exception) {
+    try {
+      boolean value = caller.call(dataTieringManager, path);
+      if (exception != null) {
+        fail("Expected DataTieringException to be thrown");
+      }
+      assertEquals(value, expectedResult);
+    } catch (DataTieringException e) {
+      if (exception == null) {
+        fail("Unexpected DataTieringException: " + e.getMessage());
+      }
+      assertEquals(exception.getMessage(), e.getMessage());
+    }
+  }
+
+  private void testDataTieringMethodWithKey(DataTieringMethodCallerWithKey caller,
+    BlockCacheKey key, boolean expectedResult, DataTieringException exception) {
+    try {
+      boolean value = caller.call(dataTieringManager, key);
+      if (exception != null) {
+        fail("Expected DataTieringException to be thrown");
+      }
+      assertEquals(value, expectedResult);
+    } catch (DataTieringException e) {
+      if (exception == null) {
+        fail("Unexpected DataTieringException: " + e.getMessage());
+      }
+      assertEquals(exception.getMessage(), e.getMessage());
+    }
+  }
+
+  private void testDataTieringMethodWithPathExpectingException(
+    DataTieringMethodCallerWithPath caller, Path path, DataTieringException exception) {
+    testDataTieringMethodWithPath(caller, path, false, exception);
+  }
+
+  private void testDataTieringMethodWithPathNoException(DataTieringMethodCallerWithPath caller,
+    Path path, boolean expectedResult) {
+    testDataTieringMethodWithPath(caller, path, expectedResult, null);
+  }
+
+  private void testDataTieringMethodWithKeyExpectingException(DataTieringMethodCallerWithKey caller,
+    BlockCacheKey key, DataTieringException exception) {
+    testDataTieringMethodWithKey(caller, key, false, exception);
+  }
+
+  private void testDataTieringMethodWithKeyNoException(DataTieringMethodCallerWithKey caller,
+    BlockCacheKey key, boolean expectedResult) {
+    testDataTieringMethodWithKey(caller, key, expectedResult, null);
+  }
+
+  @Test
+  public void testColdDataFiles() {
+    Set<BlockCacheKey> allCachedBlocks = new HashSet<>();
+    for (HStoreFile file : hStoreFiles) {
+      allCachedBlocks.add(new BlockCacheKey(file.getPath(), 0, true, BlockType.DATA));
+    }
+
+    try {
+      Set<String> coldFilePaths = dataTieringManager.getColdDataFiles(allCachedBlocks);
+      assertEquals(1, coldFilePaths.size());
```

Comment:
We should explain better why we only get 1 here, even though we say there are two non-hot files in the javadoc header. 

Also, shouldn't we assert on the exact cold file, just to make sure our logic does mark the file we know is cold?

Reply:
Yeah, good idea. Maybe I write another test to stress on cold data